uswds/uswds
# USWDS - Header: fix mobile menu shaded display

## Description
**Fix mobile menu appearance for different CSS layouts**. The menu now appears properly on layouts using flex or CSS grid. Closes #4409.

## Additional information
Removed z-index from `usa-header` in 9c0653c. There's no `position` specified, so it doesn't make sense to have it by default. Removing this fixes the display issue because the value was **lower** than the nav overlay's `z-index`.

**Other changes**
- Use consistent angle brackets for consistency (previously square brackets) -  d805212525ed5bcc05257ad67783316caeb1fa26
- Allow individual page layout templates to accept overrides — required for adding a navigation to Documentation template **without** affecting the rest of the page layout templates - e6f1817ec4512752385eb2d53b406ce439986cda

## How to test
1. Visit [Documentation page template](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/jm-mobile-menu-shaded/iframe.html?args=&id=pages-documentation-page--documentation-page&viewMode=story)
2. Verify nav is present
3. Open inspector
4. Add `style="display: flex; flex-direction: column;` to `div#root` 
5. Confirm navigation works on desktop
6. Resize to get mobile menu
7. Click on **Menu** button
8. Confirm navigation displays properly

Also, didn't notice any side effects on the other nav variations but worth double-checking.

**Browsers tested**
- Edge v103
- Firefox v102
- Safari v15 on OSX Monterey via Browserstack

---

Before you hit Submit, make sure you’ve done whichever of these applies to you:
